lib/storage/temporary-cloud-storage.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/**
+ * ADOBE CONFIDENTIAL
+ * ___________________
+ *
+ *  Copyright 2021 Adobe
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Adobe and its suppliers, if any. The intellectual
+ * and technical concepts contained herein are proprietary to Adobe
+ * and its suppliers and are protected by all applicable intellectual
+ * property laws, including trade secret and copyright laws.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Adobe.
+ */
+
+'use strict';
+
+const { v4: uuidv4 } = require("uuid");
+const filesLib = require('@adobe/aio-lib-files'); // for temp cloud storage
+
+/**
+ * This class handles temporary cloud storage
+ * This is a duplicate of asset-compute-pipeline temporary cloud storage,
+ * but as per the existing practice duplicated into SDK and will be refactored
+ * when we move to pipeline
+ */
+class TemporaryCloudStorage {
+    async _init() {
+        console.log("Using ootb storage for ('aio-lib-files')");
+        this.aioLibFiles = await filesLib.init();
+        console.log("Temporary cloud storage initialized ('aio-lib-files')");
+    }
+
+    /**
+     * Create a presigned url (read, write, delete and combinations are supported)
+     * @param {String} localFilePath local filesystem path to the file for which a presigned URL must be generated
+     * @param {String} permissions permissions (`r` for read, `w` for write, `d` for delete and their combinations)
+     * @param {Number} expiryInSeconds how long the generated presigned url will be valid
+     * @returns {String} a presigned url
+     */
+    async generatePresignURL(cloudPath, permissions = "rwd", expiryInSeconds=3600){
+        if(!this.aioLibFiles){
+            await this._init();
+        }
+
+        const presignedUrlConfig = { permissions: permissions, expiryInSeconds: expiryInSeconds };
+        const presignedUrl = await this.aioLibFiles.generatePresignURL(cloudPath, presignedUrlConfig);
```

Comment:
we want to retry if the generatingPresignedURL fails.

Reply:
(We want to retry in case there are transient errors, which is not unexpected in cloud environment. Limit the retry to max. 5 attempts. If it fails 5 times, the error encountered most likely isn't transient)

Reply:
The retry logic should be there `generatePresignURL` assuming it's using TVM and also upload and download.

Reply:
Added retry at storage.js. And added mock tests

Reply:
@bobvanmanen Are you saying we don't need to do explicit retry, I did not find retry within aioLib.

Reply:
For now we have retry for 5 times, resolving this if we need to remove we can remove later